pkg/ccl/backupccl/restore_planning.go

```diff
@@ -858,23 +876,33 @@ func allocateDescriptorRewrites(
 	// Generate new IDs for the schemas, tables, and types that need to be
 	// remapped.
 	for _, desc := range descriptorsToRemap {
-		newTableID, err := catalogkv.GenerateUniqueDescID(ctx, p.ExecCfg().DB, p.ExecCfg().Codec)
+		id, err := catalogkv.GenerateUniqueIDGreaterThanID(ctx, p.ExecCfg().DB, p.ExecCfg().Codec, maxDescIDInBackup)
```

Comment:
As discussed offline, I don't understand the need for adding this helper method/changing the way we generate IDs. This logic is all very core to restore so unless I'm missing a reason for us to change it, I'd prefer not to alter it.